backend/src/main/java/com/zpi/backend/game/GameService.java

```diff
@@ -100,4 +105,33 @@ public void acceptGame(Authentication authentication, long id) throws GameDoesNo
     public long getAmount(){
         return gameRepository.count();
     }
+
+    public ResultsDTO<UserWithGameOpinionDTO> getUsersAndGameInstancesWithGame(long gameId, double latitude, double longitude,
+                                                                               int page, int size) {
+        Pageable pageable = PageRequest.of(page, size);
+        Page<Object[]> objectsPage = gameRepository.getAllUsersWithGameAndRating(gameId, latitude, longitude, pageable);
+        List<UserWithGameOpinionDTO> userWithGameDTOPage =
+                objectsPage.stream()
+                        .map(this::convertToDTO)
+                        .toList();
+        return new ResultsDTO<>(userWithGameDTOPage,
+                new Pagination(objectsPage.getTotalElements(), objectsPage.getTotalPages()));
+    }
+
+    private UserWithGameOpinionDTO convertToDTO(Object[] columns) {
```

Comment:
can't it be done without using plain object? it's not very readable, if not maybe add comment what is in each column

Reply:
I'll leave comments here. I've tried use a DTO in query, but there was a problem with converting it to custom class.